src/routes/ChatRoom/ChatPost/ChatPost.js

```diff
@@ -0,0 +1,254 @@
+    }
+    event.stopPropagation()
+    return true
+  })
+
+  const removePostWithConfirm = useCallback((event) => {
+    if (window.confirm('Are you sure you want to remove this post? You cannot undo this.')) {
+      dispatch(removePost(id, group.slug))
+    }
+    event.stopPropagation()
+    return true
+  })
+
+  const actionItems = filter(item => isFunction(item.onClick), [
+    // { icon: 'Pin', label: pinned ? 'Unpin' : 'Pin', onClick: pinPost },
+    // { icon: 'Copy', label: 'Copy Link', onClick: copyLink },
+    { icon: 'Replies', label: 'Reply', onClick: openPost },
+    { icon: 'Edit', label: 'Edit', onClick: isCreator ? editPost : null },
+    { icon: 'Flag', label: 'Flag', onClick: !isCreator ? () => { setFlaggingVisible(true) } : null },
+    { icon: 'Trash', label: 'Delete', onClick: isCreator ? deletePostWithConfirm : null, red: true },
+    { icon: 'Trash', label: 'Remove From Group', onClick: !isCreator && canModerate ? removePostWithConfirm : null, red: true }
+  ])
+  const myEmojis = myReactions ? myReactions.map((reaction) => reaction.emojiFull) : []
+
+  const commenterAvatarUrls = commenters.map(p => p.avatarUrl)
+
+  return (
+    <Highlight {...highlightProps}>
+      <div styleName='container' ref={ref} onClick={openPost} className={className} {...bindLongPress()}>
+        <div styleName='action-bar'>
+          {actionItems.map(item => (
+            <Button
+              key={item.label}
+              noDefaultStyles
+              borderRadius={'0'}
+              onClick={item.onClick}
+              className={styles['action-item']}
+            >
+              <Icon name={item.icon} />
+            </Button>
+          ))}
+          <EmojiPicker
+            className={styles['action-item']}
+            handleReaction={handleReaction}
+            handleRemoveReaction={handleRemoveReaction}
+            myEmojis={myEmojis}
+          />
+          {flaggingVisible && <FlagContent
+            type='post'
+            linkData={{ id, slug: group.slug, type: 'post' }}
+            onClose={() => setFlaggingVisible(false)}
+          />}
+        </div>
+
+        {post.header && (
+          <>
+            <div styleName='header'>
+              <div styleName='author'>
+                <Avatar url={personUrl(creator.id)} avatarUrl={creator.avatarUrl} className={styles.avatar} />
+                <div styleName='name'>{creator.name}</div>
+              </div>
+              <div styleName='date'>{moment(createdAt).format('h:mm a')}</div>
+            </div>
+          </>
+        )}
+        {details && (
+          <ClickCatcher groupSlug={group.slug}>
+            {editing
+              ? <HyloEditor
+                contentHTML={details}
+                groupIds={post.groups.map(g => g.id)}
+                // onAddTopic={handleAddTopic}
+                onEscape={handleEditCancel}
+                onEnter={handleEditSave}
+                // onUpdate={handleDetailsUpdate}
```

Comment:
Would have thought chats still need this tho?

Reply:
Same here I think. This is a callback which in the PostEditor context is going to be used to scan entry as it's happening, and in the Chat entry context it's evidently not needed. Delete line.